examples/threading/threading.c

```diff
@@ -0,0 +1,71 @@
+#include "threading.h"
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <time.h>
+#include <errno.h>
+
+// Optional: use these functions to add debug or error prints to your application
+#define DEBUG_LOG(msg, ...)
+// #define DEBUG_LOG(msg,...) printf("threading: " msg "\n" , ##__VA_ARGS__)
+#define ERROR_LOG(msg, ...) printf("threading ERROR: " msg "\n", ##__VA_ARGS__)
+
+void *threadfunc(void *thread_param)
+{
+    struct thread_data *thread_func_args = (struct thread_data *)thread_param;
+
+    struct timespec ts;
+    int ret;
+
+    // Wait for desired milliseconds before acquiring mutex
+    ts.tv_sec = thread_func_args->m_wait_to_obtain_ms / 1000;
+    ts.tv_nsec = (thread_func_args->m_wait_to_obtain_ms % 1000) * 1000000;
+    ret = nanosleep(&ts, NULL);
+    if (ret != 0)
+    {
+        ERROR_LOG("Nanosleep Error");
+    }
+
+    // Acquire mutex and block if not available
+    pthread_mutex_lock(thread_func_args->m_mutex);
+
+    // Wait for desired milliseconds before releasing mutex
+    ts.tv_sec = thread_func_args->m_wait_to_release_ms / 1000;
+    ts.tv_nsec = (thread_func_args->m_wait_to_release_ms % 1000) * 1000000;
+    ret = nanosleep(&ts, NULL);
+    if (ret != 0)
+    {
+        ERROR_LOG("Nanosleep Error");
+    }
+
+    // Release the mutex
+    pthread_mutex_unlock(thread_func_args->m_mutex);
+
+    // Mark the thread completed
+    thread_func_args->thread_complete_success = true;
```

Comment:
Did the thread complete successfully if the sleep didn't sleep for the expected time?